Tests/KnitTests/MainActorTests.swift

```diff
@@ -0,0 +1,180 @@
+        await customGlobalActorClass.value.sayHello()
+    }
+
+    func askActorAToSayHello() async -> String {
+        await actorA.sayHello()
+    }
+
+}
+
+private final class TestAssembly: AutoInitModuleAssembly {
+
+    typealias TargetResolver = TestResolver
+
+    func assemble(container: Container<TargetResolver>) {
+
+        container.register(
+            ActorA.self,
+            mainActorFactory: { @MainActor resolver in
+                ActorA()
+            }
+        )
+
+        container.register(
+            MainClassA.self,
+            mainActorFactory: { @MainActor resolver in
+                MainClassA()
+            }
+        )
+
+        container.register(
+            Future<CustomGlobalActorClass, Never>.self,
+            mainActorFactory: { @MainActor resolver in
+                let mainClassA = resolver.unsafeResolver.resolve(MainClassA.self)!
+
+                return Future<CustomGlobalActorClass, Never>() { promise in
+                    let customGlobalActorClass = await CustomGlobalActorClass(
+                        mainClassA: mainClassA
+                    )
+                    promise(.success(customGlobalActorClass))
+                }
+            }
+        )
+
+        container.register(
+            Future<AsyncInitClass, Never>.self,
+            mainActorFactory: { @MainActor resolver in
+                return Future<AsyncInitClass, Never>() { promise in
+                    promise(.success(await AsyncInitClass()))
+                }
+            }
+        )
+
+        container.register(
+            FinalConsumer.self,
+            mainActorFactory: { @MainActor resolver in
+                let actorA = resolver.unsafeResolver.resolve(ActorA.self)!
+                let mainClassA = resolver.unsafeResolver.resolve(MainClassA.self)!
+                let customGlobalActorClass = resolver.unsafeResolver.resolve(Future<CustomGlobalActorClass, Never>.self)!
+                let asyncInitClass = resolver.unsafeResolver.resolve(Future<AsyncInitClass, Never>.self)!
+                return FinalConsumer(
+                    actorA: actorA,
+                    mainClassA: mainClassA,
+                    customGlobalActorClass: customGlobalActorClass,
+                    asyncInitClass: asyncInitClass
+                )
+            }
+        )
+
+    }
+
+    init() {}
+
+    static var dependencies: [any Knit.ModuleAssembly.Type] { [] }
+
+}
+
+class MainActorTests: XCTestCase {
+
+    @MainActor
+    func testAssembly() throws {
+        let assembler = ModuleAssembler(
+            [TestAssembly()]
+        )
+        let finalConsumer = try XCTUnwrap(assembler.resolver.resolve(FinalConsumer.self))
+
+        let asyncExpectation = expectation(description: "async task")
+
+        Task {
+            let result = await finalConsumer.askCustomGlobalActorClassToSayHello()
+            XCTAssertEqual(result, "Hello")
+            asyncExpectation.fulfill()
+        }
```

Comment:
Would it be better for the test to be async and await the result without having to create a Task?

Reply:
Updated